src/enable_variable_ops/ngraph_enter_in_catalog.cc

```diff
@@ -0,0 +1,145 @@
+/*******************************************************************************
+ * Copyright 2019 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#pragma once
+
+#include "tensorflow/core/graph/algorithm.h"
+#include "tensorflow/core/graph/graph.h"
+
+#include "ngraph/ngraph.hpp"
+#include "ngraph/serializer.hpp"
+#include "ngraph_catalog.h"
+#include "ngraph_log.h"
+#include "ngraph_utils.h"
+
+using namespace std;
+namespace ng = ngraph;
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+// Used by Variable and other Modifier Ops (NGraphVariable, NGraphAssign)
+// for accessing the variable object from resource manager using shared
+// name
+// If the op is not of type NGraphVariable,
+//    then recurse over its 1st input till we get reach the variable
+// Assumes: the Variable that is being modified is the 1st input and the only
+// modifiable input
+// If the op has many such inputs, this function needs to be called for each of
+// them
+// It is bound to terminate as the modifier ops like Assign, AssignAdd,
+// ApplyGradientDescent, etc
+// always operate on a Variable
+Status GetSharedName(Node* node, string* shared_name) {
+  if (node->type_string() == "NGraphVariable") {
+    TF_RETURN_IF_ERROR(GetNodeAttr(node->attrs(), "shared_name", shared_name));
+    if (shared_name->empty()) {
+      (*shared_name) = node->name();
+    }
+    return Status::OK();
+  }
+
+  Node* input_0;
+  TF_RETURN_IF_ERROR(node->input_node(0, &input_0));
```

Comment:
doubt: so we know this number is always `0`? Or atleast till now what you have encountered, the `0`th input is what we want to use.

Reply:
For the ops that we have seen till now, this is the case. I have this assumption mentioned in the comments